### ISSUE TITLE
zip support

### ISSUE THREAD

**Issue by nkostadinov** [rolandstarke/node-cache-manager-fs-hash#4]
Nice extension :). It would be really nice if this could zip/unzip out of the box. Do you plan on implementing such a feature.

**Comment by rolandstarke** [comment]
Hey, thank you :)

I will try.

**Comment by nkostadinov** [comment]
Actually I was using https://github.com/hotelde/node-cache-manager-fs but later found your extension which does all I needed - hash by URL, separate them by folders and play well with cluster. I'm working with big amounts of data so gzip will save a lot of space.

I looked at the sources the node-cache-manager-fs (which supports it) and I'm not sure how to implement it without breaking stuff. May be gzipping the whole file when writing and unzipping before passing it to the cb, or using external file store in json-file-store and save the zipped content to .bin file ...

**Comment by rolandstarke** [comment]
Mm so I tried to add it in v0.0.8. You can use it like:

```
const cacheManager = require('cache-manager');
const fsStore = require('cache-manager-fs-hash');

const diskCache = cacheManager.caching({
    store: fsStore,
    options: {
        zip: true,
    }
});
```

It uses `zlib.deflate` and `zlib.unzip` for the whole file.
If you use the zip option no previously cached file will be used, they have a new file ending ".gz".

Mm when you come from node-cache-manager-fs one thing to note is that they support `maxsize` and this lib does not.

**Comment by nkostadinov** [comment]
Wow man, that was fast :). Thank you and keep up the good work. So far it is working perfect.

I know about the maxsize, but it doesnt work well. It keeps the cache meta data in the memory so when you restart the server it starts from the beginning leaving garbage behind. I ended up with x12 storage that I configured in maxsize option. Hashing the key(url in my case) is simple and seems to work pretty well.
